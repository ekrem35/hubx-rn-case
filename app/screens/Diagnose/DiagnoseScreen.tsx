import React from 'react';

import {SafeAreaView} from 'react-native';

import {INavigation} from '../../lib/types';
import Button from 'components/Button';

type Props = {
  navigation: INavigation;
};

export default function DiagnoseScreen(props: Props) {
  return (
    <SafeAreaView
      style={{
        flex: 1,
        backgroundColor: '#a139Ff50',
        padding: 24,
        justifyContent: 'flex-end',
      }}>
      <Button onPress={() => props.navigation.navigate('WelcomeScreen')}>
        Go to onboarding
      </Button>
    </SafeAreaView>
  );
}
