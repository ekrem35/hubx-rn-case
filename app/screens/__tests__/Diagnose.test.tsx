/**
 * @format
 */

import 'react-native';
import React from 'react';
import DiagnoseScreen from '../Diagnose/DiagnoseScreen';

import {it} from '@jest/globals';

import {render, cleanup} from '@testing-library/react-native';

afterEach(cleanup);

describe('test diagnose screen', () => {
  it('renders correctly', async () => {
    const screen = await render(<DiagnoseScreen />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
