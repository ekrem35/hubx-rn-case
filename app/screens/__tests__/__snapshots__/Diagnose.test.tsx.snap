// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test diagnose screen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#a139Ff50",
      "flex": 1,
      "justifyContent": "flex-end",
      "padding": 24,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#28AF6E",
        "borderRadius": 12,
        "opacity": 1,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        [
          {
            "alignItems": "center",
            "color": "#FFFFFF",
            "display": "flex",
            "fontFamily": "Rubik",
            "fontSize": 15,
            "fontWeight": "700",
            "letterSpacing": -0.24,
            "lineHeight": 24,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      Go to onboarding
    </Text>
  </View>
</RCTSafeAreaView>
`;
